lib/player-page-utils.ts

```diff
@@ -0,0 +1,24 @@
+import { getImageDimensions } from './image-dimensions';
+import { getImageBaseUrl } from './urlutils';
+import { HOST_URL } from '../constants';
+
+export type Props = {
+  playbackId: string;
+  shareUrl: string;
+  poster: string;
+  aspectRatio?: number;
```

Comment:
nit (non-blocking): To make type inference easier "downstream", we could split this into types:
```ts
type BaseProps = {
  playbackId: string;
  shareUrl: string;
  poster: string;
};

type ExistingVideoProps = {
  aspectRatio: number;
  videoExists: true;
};

type NonExistingVideoProps = {
  videoExists: false;
};

export type Props = BaseProps & (ExistingVideoProps | NonExistingVideoProps);
```

Reply:
I like that!